src/iconsManifest/supportedExtensions.ts

```diff
@@ -1855,7 +1855,8 @@ export const extensions: IFileCollection = {
     },
     {
       icon: 'go',
-      extensions: [],
+      extensions: ['go.work'],
```

Comment:
I think we should create a specific entry for go.work. It's not a normal go file. We have specific go icons for `go.mod` for instance. I think we could just reuse one of the colored go extra icons to make this work.